io/net/htx_nic_devices.py

```diff
@@ -84,11 +71,32 @@ def setUp(self):
                     networkinterface.save(ipaddr, self.netmask)
                 networkinterface.bring_up()
         self.host_distro = distro.detect()
-        self.login(self.peer_ip, self.peer_user, self.peer_password)
+        self.session = Session(self.peer_ip, user=self.peer_user,
+                               password=self.peer_password)
+        if not self.session.connect():
+            self.cancel("failed connecting to peer")
         self.remotehost = RemoteHost(self.peer_ip, self.peer_user,
                                      password=self.peer_password)
+
         self.get_peer_distro()
 
+    def get_peer_distro(self):
+        """
+        Get the distro installed on peer lpar
+        """
+        cmd = "cat /etc/os-release"
+        output = self.session.cmd(cmd)
+        output = output.stdout.decode("utf-8")
```

Comment:
may use `genio` avocado library for file processing

Reply:
ah it is happnig in peer , may be we can extend  intrafce.py to have this logic which will help any file parsing in peer

Reply:
@vaishnavibhat https://github.com/avocado-framework/avocado/blob/f9ab3aaff20d5da1312984c9cb5d95b0651164fd/avocado/utils/distro.py#L298